src/mewbot/bot.py

```diff
@@ -4,28 +4,38 @@
 
 import asyncio
 import itertools
+import logging
+import signal
 
 from typing import Any, Dict, List, Optional, Set, Type
 
 from mewbot.behaviour import Behaviour
 from mewbot.data import DataSource
-from mewbot.io import (
-    IOConfig,
-    Input,
+from mewbot.core import (
+    IOConfigInterface,
+    InputInterface,
     InputEvent,
-    Output,
+    OutputInterface,
     OutputEvent,
     InputQueue,
     OutputQueue,
 )
 
+logging.basicConfig(level=logging.INFO)
+
 
 class Bot:
     name: str  # The bot's name
-    io: List[IOConfig]  # Connections to bot makes to other services
+    io_configs: List[IOConfigInterface]  # Connections to bot makes to other services
     behaviours: List[Behaviour]  # All the things the bot does
     datastores: Dict[str, DataSource[Any]]  # Data sources and stores for this bot
 
+    def __init__(self) -> None:
+        self.name = "not_set"
```

Comment:
Apparently unused unless I cannot find where it's used.

Reply:
Field is included as per spec (https://github.com/mewler/mewbot/tree/mainline/design-docs#bot) but doesn't actually really do anything -- at least until building out the UI and API structures starts happening.